projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/monitoring/DataJobMetrics.java

```diff
@@ -173,6 +172,33 @@ public void updateTerminationStatusGauge(final DataJob dataJob) {
         }
     }
 
+    /**
+     * Creates a "taurus.datajob.termination.status.counter" counter for the specified
+     * data job and execution status, if one does not exist. If a counter already
+     * exists, its value is incremented.
+     *
+     * @param dataJob The data job for which to increment the counter.
+     * */
+    public void incrementTerminationStatusCounter(final DataJob dataJob) {
+        Objects.requireNonNull(dataJob);
+
+        try {
+            var dataJobName = dataJob.getName();
+            var dataJobStatus = dataJob.getLastExecutionStatus().getAlertValue().toString();
+            var jobStatusKey = dataJobName + "__" + dataJobStatus;
+            var counter = statusCounters.getOrDefault(jobStatusKey, null);
+            var tags = createStatusCounterTags(dataJob);
+
+            if (counter == null) {
+                statusCounters.computeIfAbsent(jobStatusKey, name -> createTerminationStatusCounter(dataJobName, tags, dataJobStatus));
+            } else {
+                counter.increment();
+            }
```

Comment:
Seems like off-by one error - are we incrementing the counter the first time ?

 But I may not be seeing something. The DataJobTerminationStatusIT should be able to verify this case

Reply:
Good catch. Yes, the counter was off-by one.